### ISSUE TITLE
Add optional Overworld Inventory support

### ISSUE THREAD

**Issue by Dilapidated** [TheLX5/Powerups#45]
**Is your feature request related to a problem? Please describe.**
It's not so much a problem as it is an enhancement vs using L/R for switching between powerups. I want something similar to the NSMB inventory system for storing and switching between powerups as it would fit perfectly with some mod hacks.

**Describe the solution you'd like**
I love all the new powerups and would like a more 'authentic' way for users to store them and switch between them on the overworld screen.

**Describe alternatives you've considered**
The current switching using L/R is sufficient and works fine, this is a mere suggestion for use in a rom hack. There is another mod that does this but doesn't work with the powerups and doesn't support SA-1 either. https://www.smwcentral.net/?p=section&a=details&id=10629

**Additional context**
Thank you for creating this and all the hardwork you put into it. I realize this will probably be low priority as there's more pressing things but I hope you'll consider this feature request.

**Comment by TheLX5** [comment]
I'll see what I can do, I wish I could say that the author should take care of this, but he's been inactive for a while. The patch/mod itself is quite buggy and it might need a bit of help from me (or someone else).

**Fun fact:** Instead of having the current debug feature that changes powerups in the map by pressing L/R, I tried making a menu using the NSMBWii Inventory as a base. I ran into a couple of bugs and forgot about it since I was trying to release v3.0.0.

**Comment by Dilapidated** [comment]
Nice, didn't realize you had previously worked on a NSMBwii inventory in the past. I really like that inventory style a lot myself too, better than the smb3 style. I think something like that would be a perfect fit for this hack. I envision later down the line it could also lead to other neat things like mushroom houses, map battle bosses, puzzles, etc. for item box rewards being added to the inventory.

**Comment by TheLX5** [comment]
Alright, finally took a look at the patch and things aren't looking well.

Read my comment [here](https://www.smwcentral.net/?p=section&a=details&id=10629#comment-30929) to get a detailed explanation on what's going on.

tl;dr: I won't do this, it requires way too much time and I want to focus on other projects.
